pajbot/bot.py

```diff
@@ -130,16 +120,44 @@ def __init__(self, config: cfg.Config, args: argparse.Namespace) -> None:
         self.app_token_manager = AppAccessTokenManager(self.twitch_id_api, RedisManager.get())
         self.twitch_helix_api: TwitchHelixAPI = TwitchHelixAPI(RedisManager.get(), self.app_token_manager)
 
-        self.bot_user_id = self.twitch_helix_api.get_user_id(self.nickname)
+        # bot
+        if "bot_id" in config["main"]:
+            self.bot_user_id = config["main"]["bot_id"]
+            self.nickname = self.twitch_helix_api.get_login(self.bot_user_id)
+        elif "nickname" in config["main"]:
+            self.nickname = config["main"].get("nickname", "pajbot")
+            self.bot_user_id = self.twitch_helix_api.get_user_id(self.nickname)
+
+        # streamer
+        if "streamer_id" in config["main"]:
+            self.streamer_user_id = config["main"]["streamer_id"]
+            self.streamer = self.broadcaster = self.twitch_helix_api.get_login(self.streamer_user_id)
```

Comment:
You should create a function `require_login`, similar to `require_user_id` and then use it here and in all other places where you don't want `None` to be returned

Reply:
Is it worth making the same thing for the display name?